### PR TITLE
Fix for `Change all email verification links to use tokens`

### LINKED ISSUE #3867: Change all email verification links to use tokens
## Prerequisites
* [x] Are you running the latest version?
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue? [Search issues](https://github.com/reactioncommerce/reaction/issues)

## Issue Description
Currently email verification links just rely on URL parameters to verify email addresses which opens up a security vulnerability where someone could just change the URL parameter to be any email address. The fix for this should encompass all verification emails that use this method not just the email noted in the reproduction steps.

Note from @jshimko:
> The original Meteor accounts implementation avoided that potential security concern by creating a random token that gets used in the URL and it is then used to securely identify the user upon visiting the link. So perhaps that got broken at some point and/or UI notifications were never added to make it clear when it worked, but I think that is clearly the correct/secure way to verify an email. Having an array of tokens also allows you to have expirations on specific verification URL's (a fairly common practice).

and more context about the how:
> All of the guts of the Meteor accounts/email stuff was pulled into the app (in the file you edited here) back when I rewrote all the Email.send() stuff to use Nodemailer, so I don't see any reason why we can't just fix what's there and takes the safer route with tokens. Most of the work is already done (because it definitely used to work), so it'd be a matter of fixing whatever broke and then making the UI experience a little better when someone visits the link. Probably not a lot to work out there.

> So essentially, you could replace Meteor's Accounts.urls.verifyEmail(token) with whatever custom route you want (with the token included, of course), then set up the client side route and manually process that token with Meteor's Accounts.verifyEmail(token, callback). That also has the useful side effect of automatically logging the user in (which is what used to happen with the admin verification before it was broken). In that callback, you could do whatever notification junk you want in the UI.

## Steps to Reproduce

1. export REACTION_SECURE_DEFAULT_ADMIN=true
1. reaction reset -n && reaction
1. Watch out in console window for something along the lines
1. Notice that the URL for this verification only uses URL parameters
```
  ***************************************************
            IMPORTANT! EMAIL VERIFICATION LINK

             Email sending is not configured.

    Go to the following URL to verify email: hpwgdbe3@localhost

    http://localhost:3000/account/profile/verify?email=testing%40reactioncommerce.com
    ***************************************************
```

### PR DESCRIPTION
**Pull request by prinzdezibel**
Resolves #3867  
Impact: **major**  
Type: **bugfix**

## Issue
Currently email verification links just rely on URL parameters to verify email addresses which opens up a security vulnerability where someone could just change the URL parameter to be any email address. The fix for this should encompass all verification emails that use this method not just the email noted in the reproduction steps.

## Solution/Changes
   - No longer use route with email query parameter
   - No longer use Blaze template to render the confirmation/error page
   upon click of registration URL
   - Use meteor-accounts method Accounts.verifyEmail to login upon
   registration
   - Don't verify Meteor.user email address in Meteor method
   "accounts/verifyAccount", because this is done implicitely in
   Accounts.verifyEmail

## Breaking changes
none expected

### Test 1: Secure admin account 
```
export REACTION_SECURE_DEFAULT_ADMIN=true
reaction reset -n && reaction
```
Watch out in console window for something along the lines
```
    ***************************************************
            IMPORTANT! EMAIL VERIFICATION LINK

             Email sending is not configured.

    Go to the following URL to verify email: testing@reactioncommerce.com

    http://localhost:3000/#/verify-email/vPOom9LNuwlSRGAARSg6GwZwGDDFOxHzfdJCKFZ9Fms
    ***************************************************
```
3. Admin account in `Accounts` collection and admin in `Users` collection should be created in DB, but email.verified should be `false` for both
4. You should have received an email with a verification link (token based)
5. Pasting the URL in the browser should verify the email and display a success message
6. Refreshing the same URL again should display a "verification link expired" message
7. Navigating to /account/profile/verify again should display a "Timeout" message for 2 seconds before navigating to index route.

### Test 2: Self sign in as guest
1. Register from login popover with email address and password
2. You should be logged in immediately
3. Accounts and Users document should be created in DB, but email.verified should be `false` for both
4. You should have received an email with a verification link (token based)
5. Pasting the URL in the browser should verify the email and display a success message
6. Refreshing the same URL again should display a "verification link expired" message
7. Navigating to /account/profile/verify again should display a "Timeout" message for 2 seconds before navigating to index route.

### Test 3: Marketplace
1. As an admin, enable marketplace and invite users
2. Enrollment link should be send via email (token based)
3. Pasting the URL in the browser should verify the email and display password set form.
4. After successfully setting the password, both Accounts and Users document should result in  `email.verified = true`

### CODE REVIEW

**Comment by zenweasel** [comment]
@prinzdezibel Great job on removing the Blaze template and the thorough test instructions

**Comment by zenweasel** [comment]
@prinzdezibel You've got some lint errors here too:
```
/Users/brenthoover/Projects/js/reaction/imports/plugins/core/accounts/client/containers/passwordOverlay.js
  10:1   error  `meteor/meteor` import should occur before import of `../components/updatePasswordOverlay`  import/order
  10:8   error  A space is required after '{'                                                               object-curly-spacing
  10:15  error  A space is required before '}'                                                              object-curly-spacing

/Users/brenthoover/Projects/js/reaction/server/api/core/core.js
  820:9  error  Inconsistently quoted property '_id' found  quote-props

```

File: imports/plugins/core/accounts/client/components/verifyAccount.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  render() {
+    const classNames = classnames({
+      "fa": true,
+      "fa-times-circle-o": !!this.props.error,
+      "fa-check-circle-o": !this.props.error
+    });
+    const style = {
+      color: this.props.error ? "#f33" : "#49da49",
```
[the remark this answers is not in the document]
Comment (prinzdezibel) [reply imports/plugins/core/accounts/client/components/verifyAccount.js]: done

File: imports/plugins/core/accounts/client/containers/passwordOverlay.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { Random } from "meteor/random";
 import { Reaction } from "/client/api";
 import UpdatePasswordOverlay from "../components/updatePasswordOverlay";
 import { LoginFormValidation } from "/lib/api";
+import {Meteor} from "meteor/meteor";
```
[the remark this answers is not in the document]
Comment (prinzdezibel) [reply imports/plugins/core/accounts/client/containers/passwordOverlay.js]: done

File: private/data/i18n/ar.json [hunk 3]
```diff
@@ -707,6 +707,8 @@
           "userNotFound": "اسم المستخدم غير موجود",
```
[the remark this answers is not in the document]
Comment (prinzdezibel) [reply private/data/i18n/ar.json]: done.

File: imports/plugins/core/versions/server/migrations/20_register_verify_account.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { Migrations } from "meteor/percolate:migrations";
+import { Packages } from "/lib/collections";
+
+
+// Migration file created for removing the admin role from shop manager group, and users in the group
```
[the remark this answers is not in the document]
Comment (prinzdezibel) [reply imports/plugins/core/versions/server/migrations/20_register_verify_account.js]: yes. removed

**Comment by prinzdezibel** [comment]
@jshimko Please follow up on this if you find time. thanks

**Comment by prinzdezibel** [comment]
@spencern  conflicts resolved

**Pull request merged**